### ISSUE TITLE
Add Stratigraphic collection and concepts to site-relationship

### ISSUE THREAD

**Issue by geoderekh** [geological-survey-of-queensland/vocabularies#161]
Hi All, 
We need to add stratigraphic relationship types to site-relationship, presumably as a stratigraphic collection. While the large-scale stratigraphic relationships should be in feature-relationship (may need to be added also), this can and does change locally. The stratigraphic relationships will come from distinct pairs of CPF_INTVL_RELATE_QUALS (n=23) and CPF_INTVL_RELATE_TYPES(=13), of which there are 93 in surface geology. Hopefully the number of intvl_relate_qual_code and intvl_relate_type_code can be reduced (remove near, unknown). A couple are probably observation types as well. (contact metamorphism, alteration, brecciation, fracturing). 
Can I suggest someone pre-populate a skosify xlsx with the concepts and collection, and then we give it to someone in basins to write basic definitions. 

QUALIFIER TYPE
CONF	UNDER
FAULT	NEAR
CONMET	INTRBY
ANGUNC	UNDER
CONMET	INTR
STF	HOSTBY
GRAD	UNDER
DISC	INTS
CONC	HOST
DISC	HOSTBY
CONF	OVER
SHARP	INTRBY
SHARP	OVER
	UNKN
ANGUNC	OVER
SHARP	ABUT
BREC	HOST
FAULT	HOST
IRREG	HOST
NONCON	OVER
	INTSBY
	UNDER
UNCON	OVER
SHARP	UNKN
FAULT	ABUT
CONF	HOST
FAULT	HOSTBY
ALTER	HOST
DIS	HOST
FAULT	FAULT
STB	INTERF
CONC	UNDER
CONC	INTERF
STF	HOST
PLAN	HOST
STK	INTR
BREC	INTR
CONC	UNKN
	INTR
SHARP	INTR
FRAC	ABUT
BREC	HOSTBY
FRAC	HOST
GRAD	HOSTBY
CONC	INTR
FAULT	INTR
	INTRBY
UNCON	UNDER
FRAC	HOSTBY
DISCON	UNDER
IRREG	INTR
IRREG	HOSTBY
ALTER	HOSTBY
STK	HOST
DISCON	INTR
GRAD	OVER
CONC	OVER
	HOSTBY
	FAULT
	INTS
CONC	HOSTBY
	OVER
FAULT	INTS
FAULT	OVER
FAULT	UNDER
FAULT	INTSBY
CONMET	HOST
STF	INTERF
CONC	INTRBY
CONF	INTERF
SHARP	HOSTBY
CONF	UNKN
EROSIV	OVER
	HOST
DISC	HOST
SHARP	UNDER
DISCON	OVER
IRREG	INTRBY
NONCON	UNDER
DISC	INTR
PLAN	INTR
CONF	ABUT
	NEAR
SHARP	FAULT
	ABUT
	INTERF
UNCON	INTR
IRREG	OVER
CONF	HOSTBY
STK	HOSTBY
DIS	HOSTBY
STB	HOSTBY
DISC	INTRBY

**Comment by KellyVance** [comment]
The basics of the vocabulary should be understood before the file is constructed.

The first step is to rationalise this list, replace codes with names, identify what are actual contacts vs what are observations _about_ the contact (as you mention), remove duplicates, remove redundant terms, split apart composite terms into actual contacts and contextual observations, and importantly identify which terms have a) been used, and b) been reliably used.
I'd suggest using [CGI Contact Type](http://vocabs.ga.gov.au/cgi/sissvoc/contact-type/resource?uri=http://resource.geosciml.org/classifier/cgi/contacttype) as a guide to start process.

If someone in basins can do that task and then write definitions for the remaining terms we can push it through the info-ttl-review pipeline.

**Comment by GSQ-AI** [comment]
I had a quick look at the table and did some reorganisation and removed duplicates (After the separation of the terms). There are definitely two separate concepts here, as is indicated by the grouping of duplicated 1st/2nd terms vs. the unique 1st terms. I suspect that the unique 1st terms could be separated further as well.

I tend to agree with Vance and think an SME needs to clarify the content before we dive into appending new concepts/collections into site-relationship.

1st Term | 2nd Term
-- | --
ABUT | ABUT
FAULT | FAULT
HOST | HOST
HOSTBY | HOSTBY
INTERF | INTERF
INTR | INTR
INTRBY | INTRBY
INTS | INTS
INTSBY | INTSBY
NEAR | NEAR
OVER | OVER
UNDER | UNDER
UNKN | UNKN
  |  
ALTER |  
ANGUNC |  
BREC |  
CONC |  
CONF |  
CONMET |  
DIS |  
DISC |  
DISCON |  
EROSIV |  
FRAC |  
GRAD |  
IRREG |  
NONCON |  
PLAN |  
SHARP |  
STB |  
STF |  
STK |  
UNCON |

**Comment by geoderekh** [comment]
@KellyVance @GSQ-AI The basis for this issue needs to be re-thought since we are implementing obs_is_of_feature instead of "interval sites". This means that we will need to map these to observation and result instead of site relationship types between interval sites. 
My first thought is that we will have an observation type of "stratigraphic", but will need some though on how this maps across and how we can describe the stratigraphic order and stratigraphic contact qualifiers/types. Likely one "interval site relationship" between two features at a site will need to become one stratigraphic observation for each feature at a site, but I cant think of how we will be able/if we need to link between each observation (downhole obs can use the top/bottom, but surface strat obs are more tricky-perhaps it is inferred by having ?analytes or ?result_types of upper stratigraphic contact, lower stratigraphic contact, intrudes, intruded by etc. 
Might need to brainstorm this.

**Comment by KellyVance** [comment]
I agree we need a general overhaul in how we approach this. 
First step i think is to start calling this _entity relationships_ so it applies to sites, features, intervals, samples, etc. 
The next is to add the collection of geologic/stratigraphic relationships and use the relationships tables to describe multiple relationship types
e.g. Precipice Sandstone above Bandanna Formation
 Precipice Sandstone unconformableWith Bandanna Formation etc etc etc
Then eventually (probably in stage 2 or 3) we have to ask ourselves if we rely on a numeric ordering of intervals at a site to tell us X is above Y, or whether we define and add an observation relationship table.
Definitely need to brainstorm this.

**Comment by geoderekh** [comment]
Agree, there are problems wherever we look on this one, chief amongst them is that "Precipice Sst" and "Bandanna Fm" are just free text in result, where we would need a result_is_of_feature to make the link properly. 
It seems v important, as this will be needed before we can migrate BH or SG intervals. However, given the scope of the work (brainstorming, mapping/creating intvl_quals and intvl_relates to various obs and result vocabs, etc) probably makes anything here a stage 2/later job. If we cant get it done in short order, it should probably go to DIT for discussion and then into the risk register.

**Comment by KellyVance** [comment]
On one hand it comes back to the old argument that these are all locations (of a sort) so these intervals _should_ technically be sites. But I know the proliferation of sites makes people uncomfortable, and it could cause confusion. An observation relationship table that leverages the relationship vocab may help alleviate the problem above. It may also help with overlapping intervals of different types. e.g. in a borehole a well completion interval overlaps 2 reservoir intervals, 1 of those reservoir intervals exactly matches a stratigraphic feature, the other reservoir unit encompasses 2 stratigraphic features. A relationship table could describe all these things, plus the boundary conditions between intervals of the same type (i.e. rock units).

We should do anything we can to avoid having formation names input as free text in a results table with no defined link to the feature they are meant to be describing. 

It is probably a stage 2 problem _if_ the relationships can be temporarily stored as observation_details. But it's a solution I _think_ addresses the issue if we can adequately populate the relationship vocab and set up the table correctly to account for the multiple and various relationships that may pertain to each interval.

@DavidCrosswellGSQ FYI - so you're aware of the discussion and what might be a future issue to address.

**Comment by geoderekh** [comment]
@BlakePaul is making progress on RUxs and we need to revive/resolve this issue.

**Comment by KellyVance** [comment]
@geoderekh I agree.
We already have a number of collections under https://vocabs.gsq.digital/vocabulary/site-relationships
I think we can add a little to the topological and interval relationships e.g. NEAR, and create a Geological collection for our specifically rock relationships e.g. unconformable, faulted etc.

Should this follow on for @BlakePaul once RU largely wraps up, or is there someone else with some capacity who could compare the stratigraphic relationships to the existing site list and give us the delta with appropriate definitions?
